packages/hoprd/crates/hopr-lib/src/lib.rs

```diff
@@ -73,1470 +98,958 @@ pub enum State {
+                f,
+                "heartbeat component responsible for maintaining the network quality measurements"
+            ),
+            HoprLoopComponents::Timer => write!(f, "universal timer component for executing timed actions"),
+            HoprLoopComponents::Indexing => write!(f, "initial indexing operation into the DB"),
+            HoprLoopComponents::OutgoingOnchainTxQueue => {
+                write!(f, "on-chain transaction queue component for outgoing transactions")
             }
         }
+    }
+}
 
-        pub fn status(&self) -> State {
-            self.state
-        }
-
-        pub fn version_coerced(&self) -> String {
-            String::from(constants::APP_VERSION_COERCED)
-        }
-
-        pub fn version(&self) -> String {
-            String::from(constants::APP_VERSION)
-        }
-
-        pub async fn set_alias(&self, alias: String, peer: PeerId) {
-            self.aliases.write().await.insert(alias, peer);
-        }
-
-        pub async fn remove_alias(&self, alias: &String) {
-            self.aliases.write().await.remove(alias);
-        }
-
-        pub async fn get_alias(&self, alias: &String) -> Option<PeerId> {
-            self.aliases.read().await.get(alias).copied()
-        }
-
-        pub async fn get_aliases(&self) -> HashMap<String, PeerId> {
-            self.aliases.read().await.clone()
-        }
-
-        pub async fn get_balance(&self, balance_type: BalanceType) -> errors::Result<Balance> {
-            Ok(self.chain_api.get_balance(balance_type).await?)
-        }
-
-        pub async fn get_safe_balance(&self, balance_type: BalanceType) -> errors::Result<Balance> {
-            Ok(self
-                .chain_api
-                .get_safe_balance(self.safe_module_cfg.safe_address, balance_type)
-                .await?)
-        }
+/// Main builder of the hopr lib components
+pub fn build_components<FOnReceived, FOnSent, FSaveTbf>(
+    cfg: HoprLibConfig,
+    chain_config: ChainNetworkConfig,
+    me: OffchainKeypair,
+    me_onchain: ChainKeypair,
+    db: Arc<RwLock<CoreEthereumDb<RustyLevelDbShim>>>,
+    on_acknowledgement: FOnSent,
+    on_final_packet: FOnReceived,
+    tbf: TagBloomFilter,
+    save_tbf: FSaveTbf,
+    my_multiaddresses: Vec<Multiaddr>, // TODO: needed only because there's no STUN ATM
+) -> (
+    HoprTransport,
+    HoprChain,
+    Vec<Pin<Box<dyn futures::Future<Output = HoprLoopComponents>>>>,
+)
+where
+    FOnReceived: Fn(ApplicationData) + 'static,
+    FOnSent: Fn(HalfKeyChallenge) + 'static,
+    FSaveTbf: Fn(Box<[u8]>) + 'static,
+{
+    let identity: libp2p_identity::Keypair = (&me).into();
+
+    let (network, network_events_tx, network_events_rx) =
+        build_network(identity.public().to_peer_id(), cfg.network_options);
+
+    let addr_resolver = DbPeerAddressResolver(db.clone());
+
+    let ticket_aggregation = build_ticket_aggregation(db.clone(), &me_onchain);
+
+    let contract_addrs = ContractAddresses {
```

Comment:
ChainConfig should use ContractAddreses internally perhaps

Reply:
This will be collapsed in further refactors. `chain.rs` was an adapter for `hopr-lib`, but it should reside inside the `core-ethereum-api`. On moving that away in a subsequent cleanup, this should be refactored.